docs: add tutorial for usage with Anexia Engine (#5904)
kubernetes-sigs/external-dns: docs/tutorials/anexia-engine.md

Make sure to familiarize yourself with Anexia CloudDNS and read the following documentation to know how to create a new DNS Zone:

- [Setting up CloudDNS](https://engine.anexia-it.com/docs/en/module/clouddns/getting-started/setting-up-clouddns)

### Adding a DNS Zone

1. Log in to [Anexia Engine](https://engine.anexia-it.com/).
2. Navigate to the **Domains & DNS** section and select **CloudDNS**.
3. Click on **+ Create Zone** and provide the following details:
   - **Zone Name**: Enter the domain name (e.g., `anexia-app.com`).
   - Continue through the wizard and complete the setup
4. Save the zone configuration.

> Advanced configuration can be located at the [Anexia CloudDNS Documentation](https://engine.anexia-it.com/docs/en/module/clouddns/overview)

## Creating an Anexia API Token

To use ExternalDNS with CloudDNS, you need an API token to manage DNS zones and records.
For production usage it is advised to use a **service account** to manage DNS records.
View the official [documentation how to create a service account and retrieve the respective API token](https://engine.anexia-it.com/docs/en/module/engine-core/getting-started/managing-users-and-teams/managing-users).

> Once the user is created the retrieval of the API token itself is very similar to retrieving it from your personal user:

Reviewer: ```suggestion
Once the user is created, the retrieval of the API token itself is very similar to retrieving it from your personal user:
```

Install ExternalDNS with the Anexia webhook provider:

```bash
helm repo add external-dns https://kubernetes-sigs.github.io/external-dns/
helm upgrade -i external-dns-anexia external-dns/external-dns -f external-dns-anexia-values.yaml
```

## Deploying an Example Application

### Step 1: Create a Deployment

Create a sample deployment named `web-demo`.

```bash
kubectl create deployment web-demo --image=gcr.io/google-samples/hello-app:2.0
```

### Step 2: Create a Service

Create a `Service` manifest to expose the `hello-app` application within the cluster.
It's important to include the annotation for ExternalDNS to create DNS records for the specified hostname.

Save the following content in a file named `web-demo-service.yaml`:

```yaml
apiVersion: v1
kind: Service
metadata:
  name: web-demo
  annotations:
    external-dns.alpha.kubernetes.io/hostname: web-demo.anexia-app.com
spec:
  ports:
    - port: 80
      targetPort: 8080
  selector:
    app: web-demo
```

 > Replace `web-demo.anexia-app.com` with a subdomain of your DNS zone configured in Anexia CloudDNS.

To create the service apply it with the following command:

```bash
kubectl apply -f web-demo-service.yaml
```

> The service is going to expose the application on port 80.

Reviewer: ```suggestion
The service will expose the application on port 80.
```
